spec/migrations/migrate_cvv_message_to_redirect_auth_url.rb

```diff
@@ -0,0 +1,235 @@
+        expect(payment_pending.redirect_auth_url).to eq('https://secure.payment.com/authenticate')
+
+        expect(payment_requires_auth.cvv_response_message).to be_nil
+        expect(payment_processing.cvv_response_message).to be_nil
+        expect(payment_pending.cvv_response_message).to be_nil
+      end
+    end
+
+    context 'when payments are completed' do
+      let!(:completed_payment) do
+        create(:payment,
+               cvv_response_message: nil,
+               redirect_auth_url: nil,
+               state: 'completed')
+      end
+
+      it 'does not affect completed payments (they already have nil cvv_response_message)' do
+        migration.up
+
+        completed_payment.reload
+
+        expect(completed_payment.cvv_response_message).to be_nil
+        expect(completed_payment.redirect_auth_url).to be_nil
+      end
+    end
+
+    context 'when payments have nil cvv_response_message' do
+      let!(:nil_cvv_payment) do
+        create(:payment,
+               cvv_response_message: nil,
+               redirect_auth_url: nil,
+               state: 'pending')
+      end
+
+      it 'does not migrate payments with nil cvv_response_message' do
+        migration.up
+
+        nil_cvv_payment.reload
+
+        expect(nil_cvv_payment.cvv_response_message).to be_nil
+        expect(nil_cvv_payment.redirect_auth_url).to be_nil
+      end
+    end
+
+    context 'mixed payment states' do
+      let!(:eligible_payments) do
+        [
+          create(:payment, cvv_response_message: 'https://url1.com', state: 'requires_authorization'),
+          create(:payment, cvv_response_message: 'https://url2.com', state: 'processing'),
+          create(:payment, cvv_response_message: 'https://url3.com', state: 'pending'),
+          create(:payment, cvv_response_message: 'https://url4.com', state: 'checkout'),
+          create(:payment, cvv_response_message: 'https://url5.com', state: 'failed'),
+          create(:payment, cvv_response_message: 'https://url6.com', state: 'void'),
+          create(:payment, cvv_response_message: 'https://url7.com', state: 'invalid')
+        ]
+      end
+
+      let!(:ineligible_payments) do
+        [
+          create(:payment, cvv_response_message: nil, state: 'completed'),
+          create(:payment, cvv_response_message: nil, state: 'requires_authorization')
+        ]
+      end
+
+      it 'only migrates non-completed payments with cvv_response_message' do
+        migration.up
+
+        # Check eligible payments were migrated
+        eligible_payments.each do |payment|
+          payment.reload
+          expect(payment.redirect_auth_url).to be_present
+          expect(payment.cvv_response_message).to be_nil
+        end
+
+        # Check ineligible payments were not migrated
+        ineligible_payments.each do |payment|
+          payment.reload
+          expect(payment.redirect_auth_url).to be_nil
+          expect(payment.cvv_response_message).to eq(payment.cvv_response_message)
```

Comment:
shouldn't that be 
```suggestion
          expect(payment.cvv_response_message).to be_nil
```
`cvv_response_message` is nil in the `ineligible_payments`

Reply:
Thanks for raising this, @rioug. However, it may not always be nil. It may contain a string representing whether the CVV was matched or not. Added another example here: https://github.com/openfoodfoundation/openfoodnetwork/pull/13506/commits/0cb130600c6fd369ee421e48aca3ea1a017ccb22

Reply:
Thanks @rioug for raising this. Yes, we can replace it. It's done here: https://github.com/openfoodfoundation/openfoodnetwork/pull/13506/commits/5fca97a8ee43b7f00f0d2e2e3f6b689f931f084d